jwql/tests/test_badpix_monitor.py

```diff
@@ -0,0 +1,216 @@
+from jwql.database.database_interface import FGSBadPixelQueryHistory, FGSBadPixelStats
+from jwql.instrument_monitors.common_monitors import bad_pixel_monitor
+
+
+ON_JENKINS = '/home/jenkins' in os.path.expanduser('~')
+
+
+def test_bad_map_to_list():
+    """Check that bad pixel locations are correctly retrieved from an image
+    """
+    # Create empty DQ image
+    image = np.zeros((7, 7), dtype=np.uint16)
+
+    # Set bad pixel locations
+    do_not_use = ([0, 2, 4, 6], [0, 2, 4, 6])
+    hot = ([0, 1, 3, 5], [0, 2, 4, 6])
+    rc = ([0, 2, 4, 6], [0, 1, 3, 5])
+
+    # Populate with bad pixels
+    image[do_not_use] = dqflags.pixel['DO_NOT_USE']
+    image[hot] += dqflags.pixel['HOT']
+    image[rc] += dqflags.pixel['RC']
+
+    # Call the bad pixel monitor
+    x_do_not_use, y_do_not_use = bad_pixel_monitor.bad_map_to_list(image, 'DO_NOT_USE')
+    x_hot, y_hot = bad_pixel_monitor.bad_map_to_list(image, 'HOT')
+    x_rc, y_rc = bad_pixel_monitor.bad_map_to_list(image, 'RC')
+
+    # Test
+    assert do_not_use == (y_do_not_use, x_do_not_use)
+    assert hot == (y_hot, x_hot)
+    assert rc == (y_rc, x_rc)
+
+
+def test_check_for_sufficient_files():
+    """Be sure that the file threshold values are being used correctly
+    """
+    input_files = ['file1.fits', 'file2.fits', 'file1.fits', 'file3.fits', 'file1.fits']
+    instrument = 'nircam'
+    aperture = 'NRCA1_FULL'
+    threshold = 2
+    file_type = 'darks'
+
+    files, to_run = bad_pixel_monitor.check_for_sufficient_files(input_files, instrument, aperture, threshold, file_type)
+    assert files == ['file1.fits', 'file2.fits', 'file3.fits']
+    assert to_run is True
+
+    threshold = 4
+    files, to_run = bad_pixel_monitor.check_for_sufficient_files(input_files, instrument, aperture, threshold, file_type)
+    assert files is None
+    assert to_run is False
+
+
+def test_exclude_crds_mask_pix():
+    """Test that bad pixel images are differentiated correctly
+    """
+    common_bad = ([0, 1, 2, 3, 4], [0, 1, 2, 3, 4])
+    bad1_only = ([0, 1, 3, 4], [4, 3, 1, 0])
+    bad2_only = ([3, 3, 3, 3], [0, 1, 2, 4])
+
+    bad1 = np.zeros((5, 5), dtype=np.uint8)
+    bad1[common_bad] = 1
+    bad1[bad1_only] = 2
+
+    bad2 = np.zeros((5, 5), dtype=np.uint8)
+    bad2[common_bad] = 1
+    bad2[bad2_only] = 4
+
+    # Create a mask to help with indexing
+    mask = np.zeros(bad1.shape, dtype=bool)
+    mask[bad1_only] = True
+
+    diff = bad_pixel_monitor.exclude_crds_mask_pix(bad1, bad2)
+    assert np.all(diff[mask] == 2)
+    assert np.all(diff[~mask] == 0)
+
+    # Test the reverse case
+    mask = np.zeros(bad2.shape, dtype=bool)
+    mask[bad2_only] = True
+
+    diff = bad_pixel_monitor.exclude_crds_mask_pix(bad2, bad1)
+    assert np.all(diff[mask] == 4)
+    assert np.all(diff[~mask] == 0)
+
+
+@pytest.mark.skipif(ON_JENKINS,
+                    reason='Requires access to central storage.')
+def test_locate_rate_files():
```

Comment:
Perhaps these test_locate_* functions could be moved down a bit in order to alphabetize all of the function definitions?

Reply:
Oops. Fixed this.